### ISSUE TITLE
[Bug]: [Nightly]Search failed on all shard leaders with error 'Invalid shard leader'

### ISSUE THREAD

**Issue by NicoYuan1986** [milvus-io/milvus#22855]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:4899d84
- Deployment mode(standalone or cluster):cluster
- MQ type(rocksmq, pulsar or kafka):   kafka 
- SDK version(e.g. pymilvus v2.0.0rc2):2.3.0.dev47
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

Search failed on all shard leaders with error 'Invalid shard leader'.
```
[2023-03-18T19:32:26.893Z] [2023-03-18 18:41:41 - ERROR - pymilvus.decorators]: RPC error: [search], <MilvusException: (code=1, message=fail to search on all shard leaders, err=Invalid shard leader)>, <Time:{'RPC start': '2023-03-18 18:41:41.252436', 'RPC error': '2023-03-18 18:41:41.697302'}> (decorators.py:108)
[2023-03-18T19:32:26.893Z] [2023-03-18 18:41:41 - ERROR - ci_test]: Traceback (most recent call last):
[2023-03-18T19:32:26.893Z]   File "/home/jenkins/agent/workspace/tests/python_client/utils/api_request.py", line 26, in inner_wrapper
[2023-03-18T19:32:26.893Z]     res = func(*args, **_kwargs)
[2023-03-18T19:32:26.893Z]   File "/home/jenkins/agent/workspace/tests/python_client/utils/api_request.py", line 57, in api_request
[2023-03-18T19:32:26.893Z]     return func(*arg, **kwargs)
[2023-03-18T19:32:26.893Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/orm/collection.py", line 660, in search
[2023-03-18T19:32:26.893Z]     res = conn.search(self._name, data, anns_field, param, limit, expr,
[2023-03-18T19:32:26.893Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/decorators.py", line 109, in handler
[2023-03-18T19:32:26.893Z]     raise e
[2023-03-18T19:32:26.893Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/decorators.py", line 105, in handler
[2023-03-18T19:32:26.893Z]     return func(*args, **kwargs)
[2023-03-18T19:32:26.893Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/decorators.py", line 136, in handler
[2023-03-18T19:32:26.893Z]     ret = func(self, *args, **kwargs)
[2023-03-18T19:32:26.893Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/decorators.py", line 85, in handler
[2023-03-18T19:32:26.893Z]     raise e
[2023-03-18T19:32:26.893Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/decorators.py", line 50, in handler
[2023-03-18T19:32:26.893Z]     return func(self, *args, **kwargs)
[2023-03-18T19:32:26.893Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 518, in search
[2023-03-18T19:32:26.893Z]     return self._execute_search_requests(requests, timeout, round_decimal=round_decimal, auto_id=auto_id, **kwargs)
[2023-03-18T19:32:26.893Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 487, in _execute_search_requests
[2023-03-18T19:32:26.893Z]     raise pre_err
[2023-03-18T19:32:26.893Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 478, in _execute_search_requests
[2023-03-18T19:32:26.893Z]     raise MilvusException(response.status.error_code, response.status.reason)
[2023-03-18T19:32:26.893Z] pymilvus.exceptions.MilvusException: <MilvusException: (code=1, message=fail to search on all shard leaders, err=Invalid shard leader)>
[2023-03-18T19:32:26.893Z]  (api_request.py:39)
[2023-03-18T19:32:26.893Z] [2023-03-18 18:41:41 - ERROR - ci_test]: (api_response) : <MilvusException: (code=1, message=fail to search on all shard leaders, err=Invalid shard leader)> (api_request.py:40)
```

### Expected Behavior

pass

### Milvus Log

1. link: https://jenkins.milvus.io:18080/blue/organizations/jenkins/Milvus%20Nightly%20CI/detail/master/315/pipeline/123/
2. log: [artifacts-milvus-distributed-kafka-nightly-315-pymilvus-e2e-logs.tar.gz](https://jenkins.milvus.io:18080/job/Milvus%20Nightly%20CI/job/master/315/artifact/artifacts-milvus-distributed-kafka-nightly-315-pymilvus-e2e-logs.tar.gz)
3. collection name: search_collection_pkSwnZel
4. failed time: [2023-03-18T18:41:41.916Z] [gw5] [ 62%] FAILED testcases/test_search.py::TestCollectionSearch::test_search_with_output_fields[3000-500-128-True-False]

**Comment by NicoYuan1986** [comment]
followed closely by case 'test_search_with_consistency_bounded[500-128-False-False]'
error message:
```
[2023-03-18T19:32:26.894Z] [2023-03-18 18:41:43 - ERROR - pymilvus.decorators]: RPC error: [search], <MilvusException: (code=1, message=fail to search on all shard leaders, err=fail to Search, QueryNode ID=10, reason=Search 2 failed, reason query shard(channel)  by-dev-rootcoord-dml_177_440182307606712347v1  does not exist
[2023-03-18T19:32:26.894Z]  err %!w(<nil>))>, <Time:{'RPC start': '2023-03-18 18:41:43.226069', 'RPC error': '2023-03-18 18:41:43.670123'}> (decorators.py:108)
[2023-03-18T19:32:26.894Z] [2023-03-18 18:41:43 - ERROR - ci_test]: Traceback (most recent call last):
[2023-03-18T19:32:26.894Z]   File "/home/jenkins/agent/workspace/tests/python_client/utils/api_request.py", line 26, in inner_wrapper
[2023-03-18T19:32:26.894Z]     res = func(*args, **_kwargs)
[2023-03-18T19:32:26.894Z]   File "/home/jenkins/agent/workspace/tests/python_client/utils/api_request.py", line 57, in api_request
[2023-03-18T19:32:26.894Z]     return func(*arg, **kwargs)
[2023-03-18T19:32:26.894Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/orm/collection.py", line 660, in search
[2023-03-18T19:32:26.894Z]     res = conn.search(self._name, data, anns_field, param, limit, expr,
[2023-03-18T19:32:26.894Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/decorators.py", line 109, in handler
[2023-03-18T19:32:26.894Z]     raise e
[2023-03-18T19:32:26.894Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/decorators.py", line 105, in handler
[2023-03-18T19:32:26.894Z]     return func(*args, **kwargs)
[2023-03-18T19:32:26.894Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/decorators.py", line 136, in handler
[2023-03-18T19:32:26.894Z]     ret = func(self, *args, **kwargs)
[2023-03-18T19:32:26.894Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/decorators.py", line 85, in handler
[2023-03-18T19:32:26.894Z]     raise e
[2023-03-18T19:32:26.894Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/decorators.py", line 50, in handler
[2023-03-18T19:32:26.894Z]     return func(self, *args, **kwargs)
[2023-03-18T19:32:26.894Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 518, in search
[2023-03-18T19:32:26.894Z]     return self._execute_search_requests(requests, timeout, round_decimal=round_decimal, auto_id=auto_id, **kwargs)
[2023-03-18T19:32:26.894Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 487, in _execute_search_requests
[2023-03-18T19:32:26.894Z]     raise pre_err
[2023-03-18T19:32:26.894Z]   File "/usr/local/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 478, in _execute_search_requests
[2023-03-18T19:32:26.894Z]     raise MilvusException(response.status.error_code, response.status.reason)
[2023-03-18T19:32:26.894Z] pymilvus.exceptions.MilvusException: <MilvusException: (code=1, message=fail to search on all shard leaders, err=fail to Search, QueryNode ID=10, reason=Search 2 failed, reason query shard(channel)  by-dev-rootcoord-dml_177_440182307606712347v1  does not exist
[2023-03-18T19:32:26.894Z]  err %!w(<nil>))>
[2023-03-18T19:32:26.894Z]  (api_request.py:39)
[2023-03-18T19:32:26.894Z] [2023-03-18 18:41:43 - ERROR - ci_test]: (api_response) : <MilvusException: (code=1, message=fail to search on all shard leaders, err=fail to Search, QueryNode ID=10, reason=Search 2 failed, reason query shard(channel)  by-dev-rootcoord-dml_177_440182307606712347v1  does not exist
[2023-03-18T19:32:26.894Z]  err %!w(<nil>))> (api_request.py:40)
```

**Comment by yah01** [comment]
1. Search, and proxy pulls the shard leaders cache
2. QueryCoord balances the channel
3. Proxy forwards the search request to old leader

**Comment by yah01** [comment]
Master will be fixed with #21625,
v2.2.0 has been fixed with #22961

**Comment by NicoYuan1986** [comment]
Fixed.
